sdk/profiles/src/main/java/io/opentelemetry/sdk/profiles/data/MappingData.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.profiles.data;
+
+import java.util.List;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Describes the mapping of a binary in memory.
+ *
+ * @see "pprofextended.proto::Mapping"
+ */
+@Immutable
+public interface MappingData {
+
+  /**
+   * Unique nonzero id for the mapping.
+   *
+   * @deprecated retained only for pprof compatibility.
+   */
+  @Deprecated
+  long getId();
```

Comment:
Putting a deprecated method on a brand new data interface seems strange. If this is needed for pprof compatibility, it should be used for that, and not be deprecated. Deprecation means that it would be eventually deleted, and should never be used by any new code. Clearly, there will be new code that should use this, or you wouldn't have added it to the interface. Recommend just adding javadoc around usage, rather than deprecating.

Reply:
It is indeed strange, a point I made explicitly in the text of the PR above, along with the reasons I did it. I think including the method but marking it deprecated hits a good balance between making the SDK flexible enough to be used for generating pprof style output whilst also signalling to the user that doing so is discouraged. There isn't an annotation with that exact semantic, so Deprecated is as close as I can get whilst still having wide tool support - merely using the javadoc won't be as effective in signalling intent to users as the tools won't parse it.

I fully expect this one to generate some lively discussion - it's even possible to challenge the assumption the methods are needed at all and argue that the OpenTelemetry SDK should be opinionated about only generating OTel style data and not allowing the older pprof style, which would fit with Postel's Law better. https://en.wikipedia.org/wiki/Robustness_principle. I'm happy roll with whatever the maintainers prefer, but being new to the project I don't yet have enough feel to know what that is.

Reply:
@jack-berg any preference on this one?

Reply:
I'd say as long as its still possible to generate a pprof representation without a given property we can leave it out. Including it puts the burden on the instrumentation to generate a unique id, but a pprof serializer could generate one based on the hash of the other mapping data, right?

Reply:
In pprof the mappings are stored in an array, but locations refer to them by their id, not their array index. The simplest construction is to make the id the same as the array index, in which case the distinction doesn't matter. In OTel the id is deprecated and the lookup key is explicitly the array index, so that becomes the only way to do it. Some pprof tools may read the array into a hashmap and use the id to key into it, supporting id values that are not on the index range. Constructing such pprof files won't be possible without the id, but there probably isn't a use case for it anyhow.

Reply:
Deprecated methods removed.